core/src/main/java/org/keycloak/jose/jws/JWSBuilder.java

```diff
@@ -101,13 +106,30 @@ protected void encode(Algorithm alg, byte[] data, StringBuffer encoding) {
     }
 
     public class EncodingBuilder {
+
+        public String sign(SignatureContext signer) {
+            kid = signer.getKid();
+
+            StringBuffer buffer = new StringBuffer();
```

Comment:
Could you please switch to `StringBuilder` instead of `StringBuffer` to prevent `synchronized` penalty (potentially as another issue)? The buffer is only used in a single thread.

Reply:
Good point. Rest of the class used StringBuffer that's why it ended up like that. Will change it though.